open-metadata-implementation/adapters/governance-engines-plugins/gaian-ranger-plugin/src/main/java/org/odpi/openmetadata/adapters/governanceenginesplugins/gaianrangerplugin/RangerPolicyResultFilter.java

```diff
@@ -1,100 +1,257 @@
     }
 
-    public boolean setLogicalTable(String logicalTableName, ResultSetMetaData logicalTableResultSetMetaData) {
-        logger.logDetail("Entered setLogicalTable(), logicalTable: " + logicalTableName + ", structure: " + logicalTableResultSetMetaData);
+    private static RangerUser getRangerUser(String url, String param) {
+        RestTemplate restTemplate = new RestTemplate();
+        HttpEntity<String> entity = new HttpEntity<>(getHttpHeaders());
 
+        String finalURL = MessageFormat.format(url, param);
+        logger.logDetail("getRangerUser " + finalURL);
+        ResponseEntity<String> result = restTemplate.exchange(finalURL, HttpMethod.GET, entity, String.class);
+
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
         try {
-            queryContext.setTableName(logicalTableName);
-            queryContext.setActionType("SELECT");
-            queryContext.setSchema("Gaian"); // Hardcoded for now
-
-            // This gets overwritten in setQueriedColumns, which provides a more precise list
-            // accounting for what columns are used in select & predicate.
-            List<String> columns = new ArrayList<String>();
-            int count = logicalTableResultSetMetaData.getColumnCount();
-            for (int i = 1; i <= count; i++) {
-                String str = logicalTableResultSetMetaData.getColumnName(i);
-                columns.add(str);
-            }
-            queryContext.setColumns(columns);
-            queryContext.setResourceType("COLUMN");
-            queryContext.setColumnTransformers(new ArrayList<String>());
+            return mapper.readValue(result.getBody(), RangerUser.class);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
 
-            // User would have been set below (look for OP_ID_SET_AUTHENTICATED_DERBY_USER_RETURN_IS_QUERY_ALLOWED)
-            // For now we HARDCODE to group users - until LDAP integration done.
-            Set<String> users = new HashSet<String>();
-            users.add("users");
-            queryContext.setUserGroups(users);
+    private static HttpHeaders getHttpHeaders() {
+        HttpHeaders headers = new HttpHeaders();
 
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        List<MediaType> accept = new ArrayList<>(1);
+        accept.add(MediaType.APPLICATION_JSON);
+        headers.setAccept(accept);
 
-        } catch (SQLException e) {
-            e.printStackTrace();
-        }
-        return authorizeResult; // allow query to continue (i.e. accept this logical table)
+        String base64Credentials = getBase64CredsString();
+        headers.set("Authorization", "Basic " + base64Credentials);
+        return headers;
+    }
+
+    private static String getBase64CredsString() {
+        byte[] plainCredetialBytes = "admin:admin".getBytes();
```

Comment:
We can't have hardcoded userid/password. 
The gaian servicedef can support userid/password being provided as part of initial configuration. The standard ranger practice would be to pick up from there - in fact beyond specifying the endpoint (ie of the ranger server) every other configuration element should be retrieved from there.

If we urgently need to incorporate the code into master then please open up a new issue for this -- which we should do soon

Reply:
Can we open a new issue for the authentication part? I will try to figure out a new way to specify the credentials meanwhile.

Reply:
Tracking in https://github.com/odpi/egeria/issues/447